### PR TITLE
Update documentation for memberlist migration. Added test files for each migration step.

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR updates documentation in `memberlist.libsonnet` about how to perform migration from Consul to memberlist as KV store for hash rings. PR also adds test files for each migration step.

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/mimir/memberlist.libsonnet [hunk 1]
```diff
@@ -21,17 +21,30 @@
     memberlist_ring_enabled: false,
 
     // Migrating from consul to memberlist is a multi-step process:
-    // 1) Enable multikv_migration_enabled, with primary=consul, secondary=memberlist, and multikv_mirror_enabled=false, restart components.
+    //
```
Comment (pracucci) [inline operations/mimir/memberlist.libsonnet:24]: This documentation is great, but I think it's quite difficult to find for the final user. Can we move it to `docs/sources/operators-guide/deploying-grafana-mimir/jsonnet/`?

Comment (pstibrany) [reply operations/mimir/memberlist.libsonnet:24]: Yeah, I was thinking the same. I will do that in separate PR, if that's fine.

Comment (pracucci) [reply operations/mimir/memberlist.libsonnet:24]: Agree.

Comment (pstibrany) [reply operations/mimir/memberlist.libsonnet:24]: https://github.com/grafana/mimir/pull/2166

**Pull request merged**